tron/bin/check_job_exceeding_time.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3.8
+import logging
+import sys
+
+import pytimeparse
+
+from tron.commands import cmd_utils
+from tron.commands.client import Client
+
+
+log = logging.getLogger("check_exceeding_time")
+
+
+def parse_cli():
+    parser = cmd_utils.build_option_parser()
+    parser.add_argument(
+        "--job",
+        default=None,
+        help="Check if a particular job exceeded a time to run. If unset checks all jobs",
+    )
+    parser.add_argument(
+        "--time",
+        help="This is used to specify the time that if any job exceeds will show. Defaults to 5 hours",
+        type=int,
+        dest="time_limit",
+        default=18000,
+    )
+    args = parser.parse_args()
+    return args
+
+
+def check_if_time_exceeded(job_runs, job_expected_runtime, result):
+    states_to_check = {"queued", "scheduled", "cancelled", "skipped"}
+    for job_run in job_runs:
+        if job_run.get("state", "unknown") not in states_to_check:
+            if is_job_run_exceeding_expected_runtime(
+                job_run,
+                job_expected_runtime,
+            ):
+                result.append(job_run["id"])
+    return
+
+
+def is_job_run_exceeding_expected_runtime(job_run, job_expected_runtime):
+    states_to_check = {"queued", "scheduled", "cancelled", "skipped"}
+    if (
+        job_expected_runtime is not None
+        and job_run.get(
+            "state",
+            "unknown",
+        )
```

Comment:
do we need to do this check in both places? looks like this check is also being done by the function that calls this

Reply:
I don't think so, Idk why I initially thought the state might change so I added it in both places